### PR TITLE
Issue 2479: Progress Indicator: Design is different from the Carbon Components

### PR DESCRIPTION
**Pull request by stanislavgeorgiev**
Closes carbon-design-system/carbon-components-angular#2479

#### Changelog

**Changed**
Before: 
image

After: Update icons to the [Carbon v10 Design](https://v10.carbondesignsystem.com/components/progress-indicator/usage/)
image

Before:
image

After: Fix doubled icons for the "error" state
image

### CODE REVIEW

File: src/progress-indicator/progress-indicator.component.ts [hunk 1]
```diff
@@ -29,16 +29,41 @@ import { Step } from "./progress-indicator-step.interface";
...
+							29.3z"></path>
+					</svg>
+				</ng-container>
+				<svg *ngIf="step.state.includes('error')" class="bx--progress__warning" focusable="false"
+					preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" fill="currentColor" width="16" height="16"
+					viewBox="0 0 16 16" aria-hidden="true">
+					<path d="M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z M8,14c-3.3,0-6-2.7-6-6s2.7-6,6-6s6,2.7,6,6S11.3,14,
+						8,14z"></path>
+					<path d="M7.5 4H8.5V9H7.5zM8 10.2c-.4 0-.8.3-.8.8s.3.8.8.8c.4 0 .8-.3.8-.8S8.4 10.2 8 10.2z"></path>
 				</svg>
```
Comment (Akshat55) [inline src/progress-indicator/progress-indicator.component.ts:66]: Hey Stan, could you please use the ibmIcon directive with the following icon names:

* incomplete
* checkmark--outline
* warning
* circle-dash

An example of this for `incomplete/error` state would be:
```
<svg ibmIcon="incomplete" size="16" [ngClass]="{ 'bx--progress__warning': step.state.includes('error') }"></svg>
```

When testing, if you do NOT see the icon, you will need to import them in `icon.module.ts`.

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/progress-indicator/progress-indicator.component.ts (hunk 1 above)
Comment (stanislavgeorgiev) [reply src/progress-indicator/progress-indicator.component.ts:66]: @Akshat55 I will give it a try - maybe the icons before were just not the right names

File: src/progress-indicator/progress-indicator.component.ts [hunk 2]
```diff
@@ -85,7 +81,7 @@ export class ProgressIndicator implements OnChanges {
 	}
 	private _current: number;
 
-	constructor(protected experimental: ExperimentalService) {}
+	constructor(protected experimental: ExperimentalService) { }
```
Comment (Akshat55) [inline src/progress-indicator/progress-indicator.component.ts:84]:
```suggestion
	constructor(protected experimental: ExperimentalService) {}
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Review by Akshat55 — APPROVED** [review]
Approving as there is no breaking change. Icons are updated but behavior is still the same.

File: src/progress-indicator/progress-indicator.component.ts (hunk 2 above)
Comment (stanislavgeorgiev) [reply src/progress-indicator/progress-indicator.component.ts]: This was fixed by the linter, I think

**Review by zvonimirfras — APPROVED** [review]

**Comment by stanislavgeorgiev** [comment]
@zvonimirfras @Akshat55 Do I submit a PR to `next` too or that's something that you're going to do?

**Pull request merged**